src/liballoc/boxed.rs

```diff
@@ -532,7 +524,15 @@ impl<T: ?Sized> Box<T> {
     where
         T: 'a, // Technically not needed, but kept to be explicit.
     {
-        unsafe { &mut *Box::into_raw(b) }
+        let b = mem::ManuallyDrop::new(b);
+        let unique = b.0;
+        // Box is kind-of a library type, but recognized as a "unique pointer" by
+        // Stacked Borrows.  This function here corresponds to "reborrowing to
+        // a raw pointer", but there is no actual reborrow here -- so
+        // without some care, the pointer we are returning here still carries
+        // the tag of `b`, with `Unique` permission.
+        // We round-trip through a mutable reference to avoid that.
```

Comment:
This comment needs rewording now that the result is not a raw pointer any more but a mutable reference... I can propose something later.

Reply:
I’ve reworded based on your explanation in the other sub-thread.